src/Resources/skeleton/forgottenPassword/ForgottenPasswordController.tpl.php

```diff
@@ -0,0 +1,118 @@
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+
+class <?= $class_name; ?> extends <?= $parent_class_name; ?><?= "\n" ?>
+{
+    // token sent by e-mail is valid 24 hours
+    const TOKEN_TTL_HOURS = 24;
+
+    /**
+     * @Route("/forgotten-password", name="app_forgotten_password")
+     */
+    public function request(Request $request, \Swift_Mailer $mailer): Response
+    {
+        $form = $this->createForm(<?= $username_form_class_name ?>::class);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $user = $this->getDoctrine()->getRepository(<?= $user_class_name ?>::class)->findOneBy([
+                '<?= $username_field ?>' => $form->get('<?= $username_field ?>')->getData(),
+            ]);
+
+            // do not reveal whether an user account was found or not
+            if (!$user) {
+                return $this->redirectToRoute('app_check_email');
+            }
+
+            // generate a reset password token, the user could use to change his password
+            $resetPasswordToken =$this->generateConfirmationToken();
+
+            $user->setResetPasswordToken($resetPasswordToken);
+            $user->setPasswordRequestedAt(new \DateTime());
+
+            $message = (new \Swift_Message('Your password reset request'))
+                ->setFrom(['noreply@mydomain.com' => 'Noreply'])
+                ->setTo($user-><?= $email_getter ?>())
+                ->setBody($this->renderView('forgottenPassword/email.txt.twig', [
+                    'token' => $resetPasswordToken,
+                    'tokenLifetime' => self::TOKEN_TTL_HOURS,
+                ]))
+            ;
+            $mailer->send($message);
+
+            $this->getDoctrine()->getManager()->flush();
+
+            return $this->redirectToRoute('app_check_email');
+        }
+
+        return $this->render('forgottenPassword/request.html.twig', [
+            'requestForm' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/check-email", name="app_check_email")
+     */
+    public function checkEmail()
+    {
+        return $this->render('forgottenPassword/check_email.html.twig', [
+            'tokenLifetime' => self::TOKEN_TTL_HOURS,
+        ]);
+    }
+
+    /**
+     * @Route("/reset-password/{token}", name="app_reset_password")
+     */
+    public function reset(Request $request, $token, UserPasswordEncoderInterface $passwordEncoder): Response
+    {
+        $user = $this->getDoctrine()->getRepository(<?= $user_class_name ?>::class)->findOneBy([
+            'resetPasswordToken' => $token,
+        ]);
+
+        if (!$user
+            || !$user->getPasswordRequestedAt()
+            || ($user->getPasswordRequestedAt()->getTimestamp()+self::TOKEN_TTL_HOURS*3600) <= (new \DateTime())->getTimestamp()
+        ) {
```

Comment:
I also think it might be nicer to have a method on the User here too - like `isResetPasswordTokenExpired()`